pkg/golang/build.go

```diff
@@ -0,0 +1,129 @@
+package golang
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+type Environ struct {
+	// GOARCH.
+	Arch string
+
+	// GOOS.
+	OS string
+
+	// GOPATH.
+	Gopaths []string
+
+	// GOROOT.
+	Root string
+}
+
+func Guess() Environ {
+	var env Environ
+	if p := os.Getenv("GOPATH"); p != "" {
+		env.Gopaths = append(env.Gopaths, strings.Split(p, ":")...)
+	}
+
+	if a := os.Getenv("GOARCH"); a != "" {
+		env.Arch = a
+	} else {
+		env.Arch = runtime.GOARCH
+	}
+
+	if os := os.Getenv("GOOS"); os != "" {
+		env.OS = os
+	} else {
+		env.OS = runtime.GOOS
+	}
+
+	if r := os.Getenv("GOROOT"); r != "" {
+		env.Root = r
+	} else {
+		env.Root = runtime.GOROOT()
+	}
+	return env
+}
+
+// FindPackage returns the full path to `pkg` according to the context's Gopaths.
+func (c Environ) FindPackage(pkg string) (string, error) {
```

Comment:
Wouldn't `types.ImporterFrom.ImportFrom` be more robust?

https://godoc.org/go/types#ImporterFrom

I think you're supposed to get it with `importer.For("source", nil).(types.ImporterFrom)`? bleh.

https://godoc.org/go/importer#For

Reply:
Argh, I see this is already done in a later commit.